sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -226,7 +224,7 @@ private[sql] case class ParquetRelation2(
     private var commonMetadataStatuses: Array[FileStatus] = _
 
     // Parquet footer cache.
-    var footers: Map[FileStatus, Footer] = _
+    var footers: Map[Path, Footer] = _
```

Comment:
I think the advantage to making this a `Map[FileStatus, Footer]` is that if the file changes the cache is automatically invalidated.  Though perhaps this does not matter if we are also caching those?

Also, I think that we need to bound the size of this, perhaps in some global way?